torch/_dynamo/variables/builder.py

```diff
@@ -1451,7 +1451,10 @@ def _clone_input(value):
         # tensor, the stored example value will update too!)
         example_value = _clone_input(example_value)
         proxy.node.meta["example_value"] = example_value
-        specialized_props = target_cls.specialize(example_value)
+        specialized_props = target_cls.specialize(
+            example_value,
+            test_is_contiguous=not torch._C._functorch.is_batchedtensor(example_value),
```

Comment:
Agree this is a bit sus. What's going on here?

Reply:
Calling `is_contiguous` on a BatchTensor with a memory format other than `contiguous_format` result in an exception. This code is just a switch to prevent the exception from happening. 

I also just pushed a commit that moves the check to inside `specialize` as suggested by Richard.

Reply:
How did this work before this PR when compiling vmap? Was it just not tested?

Reply:
This combination (vmap over torch.cond) didn't work before this PR. To get it to work, we need to teach Dynamo about BatchedTensor (which we have not done before). BatchedTensor doesn't have semantics for is_contiguous in certain cases.

Reply:
Oh, I thought this path would be hit already on regular `vmap`  (without `torch.cond`) already

Reply:
@lezcano in our previous vmap x torch.compile design Dynamo didn't peek into vmap (so it would never see any BatchedTensors). This ran us into subtle problems, so in our new design (that @guilhermeleobas is working on), Dynamo will actually see the BatchedTensors. Part of this PR (which says how to fake-ify a BatchedTensor) is necessary for that

Reply:
thank you for the explanation!